plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/expression/ControlFlow.kt

```diff
@@ -98,6 +98,11 @@ data class While(
             Stmt.If(condVar.toViperBuiltinType(ctx), bodyBlock, els = Stmt.Seqn(), ctx.source.asPosition)
         }
         ctx.addLabel(breakLabel.toViper(ctx))
+        invariants.forEach {
+            ctx.addStatement {
+                Stmt.Assert(it.pureToViper(toBuiltin = true))
+            }
+        }
```

Comment:
Please put a TODO here to change this back to a label invariant once we upgrade Viper